Missing dependency python-dateutil in setup.py

I noticed that `python-dateutil` is missing from setup.py in `main`.  The line "python-dateutil >= 2.7.2, <3.0dev", was removed in https://github.com/googleapis/python-bigquery/commit/76d88fbb1316317a61fa1a63c101bc6f42f23af8. 

I'm seeing failures in samples testing for other projects:
https://source.cloud.google.com/results/invocations/8e737a80-a041-468e-a2de-184d5ec1662f/log

```
******************** TESTING PROJECTS ********************
------------------------------------------------------------
- testing samples/snippets
------------------------------------------------------------
nox > Running session py-3.10
nox > Creating virtual environment (virtualenv) using python3.10 in .nox/py-3-10
nox > python -m pip install -r requirements.txt
nox > python -m pip install -r requirements-test.txt
nox > python -m pip install -e /workspace
nox > pytest --junitxml=sponge_log.xml
ImportError while loading conftest '/workspace/samples/snippets/conftest.py'.
conftest.py:23: in <module>
    from google.cloud import bigquery
.nox/py-3-10/lib/python3.10/site-packages/google/cloud/bigquery/__init__.py:35: in <module>
    from google.cloud.bigquery.client import Client
.nox/py-3-10/lib/python3.10/site-packages/google/cloud/bigquery/client.py:64: in <module>
    from google.cloud.bigquery import _job_helpers
.nox/py-3-10/lib/python3.10/site-packages/google/cloud/bigquery/_job_helpers.py:24: in <module>
    from google.cloud.bigquery import job
.nox/py-3-10/lib/python3.10/site-packages/google/cloud/bigquery/job/__init__.py:17: in <module>
    from google.cloud.bigquery.job.base import _AsyncJob
.nox/py-3-10/lib/python3.10/site-packages/google/cloud/bigquery/job/base.py:27: in <module>
    from google.cloud.bigquery import _helpers
.nox/py-3-10/lib/python3.10/site-packages/google/cloud/bigquery/_helpers.py:24: in <module>
    from dateutil import relativedelta
E   ModuleNotFoundError: No module named 'dateutil'
```

--- a/setup.py
+++ b/setup.py
@@ -42,7 +42,6 @@ dependencies = [
     "google-cloud-core >= 1.4.1, <3.0.0dev",
     "google-resumable-media >= 0.6.0, < 3.0dev",
     "packaging >= 14.3",
-    "proto-plus >= 1.10.0",  # For the legacy proto-based types.
     "protobuf >= 3.12.0",  # For the legacy proto-based types.
     "python-dateutil >= 2.7.2, <3.0dev",
     "pyarrow >= 3.0.0, < 8.0dev",

--- a/setup.py
+++ b/setup.py
@@ -44,6 +44,7 @@ dependencies = [
     "packaging >= 14.3",
     "proto-plus >= 1.10.0",  # For the legacy proto-based types.
     "protobuf >= 3.12.0",  # For the legacy proto-based types.
+    "python-dateutil >= 2.7.2, <3.0dev",
     "pyarrow >= 3.0.0, < 8.0dev",
     "requests >= 2.18.0, < 3.0.0dev",
 ]

